src/Symfony/Component/Yaml/Parser.php

```diff
@@ -237,8 +237,8 @@ private function doParse($value, $flags)
                 }
 
                 if (!(Yaml::PARSE_KEYS_AS_STRINGS & $flags) && !is_string($key)) {
```

Comment:
integer keys must be allowed, not only string keys (PHP array support integers). Can you also fix it (there is a PR fixing it currently, but together with other changes and so targetting 3.4)